### ISSUE TITLE
Selecting .Id should be invalid

### ISSUE THREAD

**Issue by TomMalow** [delegateas/XrmMockup#70]
The following context request should be invalid but is currently accepted:

```csharp
var accounts = xrm.AccountSet.Select(x => x.Id).ToList();
```
CRM throws and error as Account does not hold an "Id" attribute.

Be careful about implementation as I believe there exist some entities with such an attribute.

This might be related to a bug in XrmContext instead

**Comment by henrikhannemose** [comment]
I have plenty of `.Select(a => a.Id)` in my code, both running in plugins and in external code (which works). Let's look closer at this.

**Comment by TomMalow** [comment]
After further reading through the large error log i found a slightly extended error message that contain further detail:
> Invalid type for entity id value
> Parameter name: Target

One of my other changes was to bulk a series of updates by using the `PerformAsBulk` from XrmContext. Might be an issue related with that. Have to investigate further

**Comment by TomMalow** [comment]
After testing a bit I can conclude that selecting on `Id` does not cause issues. The issues does not seems to stem from XrmContext bulk method.

I will close this issue down and reopen later if necessary

**Comment by TomMalow** [comment]
I am repoening this issue as I found the initial problem in a related issue. It's a mix between a XrmContext bug and XrmMockup bug.

The following call is invalid in CRM but valid in XrmMockup:
```
var account = Account.Retrieve(orgAdminService, "Some_Account_Id", x => x.Id)
```
This call makes the following call in the brackground
```
var account = orgAdminService.Retrieve("account", "Some_Account_Id", new ColumnSet("id"));
```

This call will result in the following error:
>Microsoft.Crm.Metadata.AttributeMetadataNotFoundException: 'Account' entity doesn't contain attribute with Name = 'id' and NameMapping = 'Platform'

So the issue is that XrmMockup allows a retrieve call with the ColumnSet containing the attribute "id" which most entities does not have. (some entities do have this attribute).
Also, XrmContexts strongly typed retrieve call allows the user to use .id and transform it into an invalid retrieve call.

**Comment by magesoe** [comment]
Fixed in version 0.10.0
